### PR TITLE
feat(redesign): form builder redesign

### LINKED ISSUE #824: Add safe wallet dark mode support in the transaction builder
## Overview
Currently the transaction builder only supports the light mode theme, making a bad user experience for the users who has the dark mode active in the wallet, because they would be seeing dark in the wallet but white in the transaction builder screens.

## Requirements
Transaction builder should support dark mode.

### PR DESCRIPTION
**Pull request by clovisdasilvaneto**
This PR address the design reviews in the[ Design QA document](https://www.notion.so/safe-global/Tx-builder-UI-QA-11f8180fe573809d8661f0013f611313).

## What it solves
Resolves #824 

## How to test it
- open the[ Design QA document](https://www.notion.so/safe-global/Tx-builder-UI-QA-11f8180fe573809d8661f0013f611313)
- open the transaction builder
- All the requested changes should be done, except the margin of the form title and inputs.

## Screenshots
Screenshot 2024-10-18 at 10 11 14
Screenshot 2024-10-18 at 10 11 22

### CODE REVIEW

File: apps/tx-builder/src/pages/CreateTransactions.tsx [hunk 1]
```diff
@@ -110,9 +110,9 @@ const CreateTransactions = () => {
 
 export default CreateTransactions
 
-const TransactionsSectionWrapper = styled(Grid)`
+const TransactionsSectionWrapper = styled(Grid)<{ sticky?: boolean }>`
```
Comment (usame-algan) [inline apps/tx-builder/src/pages/CreateTransactions.tsx:113]: This `sticky` prop doesn't seem to be used.

File: apps/tx-builder/src/theme/SafeThemeProvider.tsx [hunk 2]
```diff
@@ -1,17 +1,24 @@
...
+import React, { useEffect, useMemo, useState, type FC } from 'react'
 import { type Theme } from '@mui/material'
 import { ThemeProvider } from '@material-ui/core'
 import createSafeTheme from './safeTheme'
 import { getSDKVersion } from '@safe-global/safe-apps-sdk'
 
+export enum EModes {
+  DARK = 'dark',
+  LIGHT = 'light',
+}
```
Comment (usame-algan) [inline apps/tx-builder/src/theme/SafeThemeProvider.tsx:10]: nit: Not sure what the E stands for. Is there a more descriptive name?

File: apps/tx-builder/src/components/CreateNewBatchCard.tsx [hunk 3]
```diff
@@ -40,65 +44,91 @@ const CreateNewBatchCard = ({ onFileSelected }: CreateNewBatchCardProps) => {
...
+            <StyledText variant="body1" error={isAcceptError}>
+              The uploaded file is not a valid JSON file
+            </StyledText>
+          ) : (
+            <>
+              <Icon type="termsOfUse" size="sm" />
+              <StyledText variant="body1">Drag and drop a JSON file or</StyledText>
+              <StyledButtonLink color="secondary" onClick={handleBrowse}>
+                choose a file
+              </StyledButtonLink>
```
Comment (usame-algan) [inline apps/tx-builder/src/components/CreateNewBatchCard.tsx:70]: From the designs it looks like this field should be white and only turn green when dragging a file over it. And the error when dragging an unsupported file should have a different shade of red:

Current:
Screenshot 2024-10-18 at 10 35 48

Design:
Screenshot 2024-10-18 at 10 33 40
Screenshot 2024-10-18 at 10 36 13


Current:
Screenshot 2024-10-18 at 10 34 59

Design:
Screenshot 2024-10-18 at 10 35 33

File: apps/tx-builder/src/theme/SafeThemeProvider.tsx (hunk 2 above)
Comment (clovisdasilvaneto) [reply apps/tx-builder/src/theme/SafeThemeProvider.tsx:10]: E is for `Enum`, it is a typescript convention. Usually `T` for types and `I` for interfaces, so when you use the type, you would already know what it is without the needing of taking a look in the definition.

File: apps/tx-builder/src/components/CreateNewBatchCard.tsx (hunk 3 above)
Comment (clovisdasilvaneto) [reply apps/tx-builder/src/components/CreateNewBatchCard.tsx:70]: Hey @usame-algan we are using the same component and colors from the web-wallet. That is why I didn't change this one.

File: apps/tx-builder/src/pages/CreateTransactions.tsx (hunk 1 above)
Comment (clovisdasilvaneto) [reply apps/tx-builder/src/pages/CreateTransactions.tsx:113]: oh true, nice catch.

**Review by usame-algan — APPROVED** [review]

**Comment by clovisdasilvaneto** [comment]
**Design Review:** ✅

**Comment by liliya-soroka** [comment]
1. the tooltip text is  hardly seen
image
image
2. when the file contains menu actions:
New design
image

Old design 
image

3. Missed info block when the list of actions is long 
image
 Old 
 
image

**Comment by liliya-soroka** [comment]
Verified

**Pull request merged**